ci: Update PHP/WP test versions

nlemoine opened the pull request:
## Issue

PHP newer versions are now more widely in WP ecosystem.

## Solution

Update tested versions.

## Impact

Waiting for the workflow to run :)

## Usage Changes
<!-- Are there are any usage changes that we need to know about? If so, list them here so that we can integrate it in the release notes and developers know what usage changes are associated to your PR.
-->

## Considerations
<!-- As we do not live in an ideal world it's worth to share your thought on how we could make the solution even better. -->


## Testing
<!-- Are unit tests included? If they need to be written, please provide pseudo code for a scenario that fails without your code, but succeeds with it -->

nlemoine committed:
ci: Update PHP/WP test versions

Levdbas reviewed (dismissed):
LGTM @nlemoine , nice to start testing newer versions!

nlemoine commented:
Thanks @Levdbas ! Don't merge it yet. I'll fix a couple of warnings and errors

nlemoine committed:
tests: Move fixtures assets

nlemoine committed:
test: Modernize test suite with PHPUnit 11.5+/12 and Mantle Testkit

Migrate the testing infrastructure to use modern tooling and conventions,
providing a solid foundation for future development.

- Switch to Mantle Testkit for WordPress integration testing
- Enables running tests without MySQL (SQLite support)
- Removes need for manual test framework maintenance
- Add PHPUnit ^11.5 || ^12 support (11.5+ for PHP 8.2, 12 for PHP 8.3+)

- Convert all doc-comment annotations to PHP 8 attributes
(@group → #[Group], @ticket → #[Ticket], etc.)
- Add #[IgnoreDeprecations] to tests using setExpectedDeprecated()
- Migrate phpunit.xml to PHPUnit 11.5 schema
- Configure ignoreIndirectDeprecations for cleaner output

- Reorganize tests into logical directories (Image/, Cache/, Integration/, Factory/)
- Rename test files to PSR-4 compliant *Test.php convention
- Add Timber\Tests namespace to all test classes
- Remove WP-CLI testsuite (functionality covered by CleanerTest.php)

- Configure ECS to enforce `use` imports over FQN in attributes
- Move phpstan config to standard /phpstan location

- mantle-framework/testkit: ^1.15 (testing framework)
- phpunit/phpunit: ^11.5 || ^12
- ergebnis/phpunit-slow-test-detector: ^2.20

Run tests: composer test (no MySQL required)

nlemoine commented:
### Testing Framework Migration
- **Switched to [Mantle Testkit](https://mantle.alley.com/docs/testing/testkit)** for WordPress integration testing
- **No MySQL required** - tests run with SQLite via [WordPress SQLite Integration](https://github.com/WordPress/sqlite-database-integration)
- Just run `composer test` and it works out of the box

### PHPUnit Modernization
- **PHPUnit 11.5+ / 12 support** (11.5 for PHP 8.2, 12 for PHP 8.3+)
- Converted all doc-comment annotations to **PHP 8 attributes**:
- `@group` → `#[Group]`
- `@ticket` → `#[Ticket]`
- `@requires` → `#[RequiresPhp]`
- Added `#[IgnoreDeprecations]` to ~75 tests that validate deprecated functionality
- Migrated `phpunit.xml` to PHPUnit 11.5 schema

### Test Organization
- Reorganized tests into logical directories (`Image/`, `Cache/`, `Integration/`, `Factory/`)
- Renamed files to PSR-4 compliant `*Test.php` convention
- Added `Timber\Tests` namespace to all test classes

### Bug Fixes
- Removed dead WP-CLI testsuite reference (functionality covered by `CleanerTest.php`)

### Code Quality
- ECS configured to enforce `use` imports over fully-qualified names
- Moved phpstan config to standard `/phpstan` location
- Added slow test detector to identify performance issues

nlemoine commented:
Hey @timber/rangers!

While working on this PR, I thought it was a good opportunity to update the whole testing suite. It was pretty old, a bit messy and needed refactoring. So I updated several parts, laying a solid foundation for moving forward safely.

You can read a complete summary of what's been done in the previous comment. Switching to https://mantle.alley.com/docs/testing/testkit adds a dependency but removes the painful testing framework maintenance and provides many features (faking HTTP requests, dealing with WP state, hooks assertions, etc.). My favorite one is probably being able to execute tests without a MySQL instance (using https://github.com/WordPress/sqlite-database-integration). That means you can run `composer test` and get tests running right away, making our lives and contributors' lives easier.

I highly recommend checking https://mantle.alley.com/docs/testing/testkit. I kept changes as minimal as possible but there are probably more opportunities to replace in-house code with Mantle Testkit features.

Key improvements for future development:
- PHPUnit 11/12 compatibility with PHP 8 attributes
- No MySQL required for local testing
- Cleaner test organization and namespacing
- Better deprecation testing support
- Improved idempotency

@gchtr, I removed the WP-CLI test by replacing it with CleanerTest.php which covers the critical part that needs testing. The WP-CLI command is just a wrapper around that class and the setup WP-CLI needs isn't worth the complexity IMHO. You'll let me know if that's ok for you.

Merry Xmas guys! 🎄

nlemoine committed:
ci: Update default PHP version to 8.2 in setup action

nlemoine commented:
Note that the failing tests for PHP 8.5 was left on purpose. I'll fix it as soon as soon this PR is merged so we can commit it as a proper fix in changelog.

nlemoine committed:
fix: Update static analysis workflow to correctly handle changed files

nlemoine commented:
Final but important missing note, this PR also bumps minimum requirements per our backward compatibility policy: PHP 8.2+ and WP 6.3+.

Levdbas commented:
Nice work @nlemoine,

This indeed gives us a solid foundation to build upon in the future. And the fact we can test more easy in the future is really nice ass well.

Stat wise we are good on bumping to 8.2 since we only have 10% running on 8.1.

<img width="510" height="497" alt="afbeelding" src="https://github.com/user-attachments/assets/bbf43d79-51da-4131-8c7a-f431cc9f78fa" />

Levdbas commented:
Hey @nlemoine , are we ready to merge this in? 📦

nlemoine commented:
I was waiting for @gchtr review 😄

But I would like to merge this pretty quick before getting too many conflicts 😉

nlemoine committed:
test: Fix ancestors post tests

gchtr commented:
LGTM! 🚀
